ethcore/evm/src/interpreter/mod.rs

```diff
@@ -273,10 +275,11 @@ impl<Cost: CostType> Interpreter<Cost> {
 		let valid_jump_destinations = None;
 		let gasometer = Cost::from_u256(params.gas).ok().map(|gas| Gasometer::<Cost>::new(gas));
 		let stack = VecStack::with_capacity(schedule.stack_limit, U256::zero());
-
+		let return_stack = Vec::new();
```

Comment:
does it make sense to use `with_capacity(MAX_SUB_STACK_SIZE)` to avoid reallocations here?

Reply:
1024 is really a lot, I think that 32 levels are enough for most of the smartcontracts there, but just to avoid magic numbers `MAX_SUB_STACK_SIZE` is ok. I suppose that there are not going to be any performance impact from 32 to `MAX_SUB_STACK_SIZE`. (?)

Reply:
yeah, it's hard to tell what the typical average subroutine stack size would be, but we already allocate stack above with `with_capacity(stack_limit)` and since we know the hard bound on the sub stack size, we could do the same here for `return_stack`. Of course w/o proper analysis it sounds like a premature optimization, it's just something to be aware of, that with `Vec::new()` we may do a lot of unnecessary reallocations (`log2(1024)`).